### DISCUSSION TITLE
Integrating AdAdmin with an existing laravel site

### DISCUSSION THREAD

**Discussion by trengram** [giuliopons/adadmin#40] [Ideas]
Hello Giulio,
I met a challenge with AdAdmin and I would wish to know how to go about it.
 
I added the scripts to an existing laravel website, which already has authentication, including signup, login, verify, password reset, etc. However, when I add AdAdmin, it comes with its own authentication system. I found it quite challenging to merge the user databases with those already created. May I kindly request your assistance with this to make everything seamless, so that a user who registered already can access the AdAdmin section as a client; and keep the administrator of the site, as the ad admin, and how to create the webmasters, who can also be existing users. 
Thank you for a wonderful AdServer and for assisting with this implementation.

**Comment by giuliopons** [comment]
Hi Starme2022,
are you sure you need users to login into AdAdmin or you just need users to create banners?
For another site, for example, I've implemented a purchase process with existing users and then I write the banner created on the AdAdmin tables/files. In that case I've linked all banners purchased by different users to the same Client and Campaign in AdAdmin.
There was no need to let the users into AdAdmin.

**Comment by trengram** [comment]
I was hoping that for users/advertisers to create ads and pay for them, they should be able to log in to the site. So, if they must log in, then they need to be those existing in the platform already, or those who join subsequently.

**Reply by giuliopons** [reply]
Ok.
I think you have to implement two mechanisms:

1) a function in your code that creates a user in AdAdmin tables, at least in `frw_utenti` (it should be enough, but - for advertisers - 
 probably you would like to create a record in `7banner_clienti` and in `frw_extrauserdata`).

2) a function in your code that login the user in AdAdmin by creating the same session variables as the AdAdmin login process (read the code in `login.class.php`), so you can log users from your main login and send users to AdAdmin home without passing from AdAdmin login.

I suggest you to **don't modify** the AdAdmin code because it will make problematic to update your installation. If you need to reuse my `session.class.php` and `login.class.php` you could create a new file, for example, `login-integration.php` file inside `/amb/src` folder that includes config and classes as performed in `login.php` file.

During the update process your file will not be overwritten.

Let me know what you think!

**Comment by trengram** [comment]
Hello @giuliopons 
I tried the first option but it didn't work. The second option is a bit complicated, mainly due to the words used in some instances. If you can have a solution to some of use who appreciate and would like to use AdAdmin with their existing laravel applications, it would go a long way. I hope for your feedback.

**Reply by giuliopons** [reply]
I'm going to clean up and translate the code for the login process
Maybe in this way it will be easy to understand how to reuse the code to make the integration.

**Reply by giuliopons** [reply]
Have you managed to create a user in AdAdmin tables from your code?

**Reply by giuliopons** [reply]
I'm modifying the login process to allow an external login... but the user must exists also in the AdAdmin tables.

**Comment by trengram** [comment]
I modified my RegisterController as follows, so it could create users in the two tables. However, it still couldn't fill in the details in the frw_utenti table.

protected function create(array $data)
{
    $user = User::create([
        'username' => $data['username'],
        'name' => $data['name'],
        'surname' => $data['surname'],
        'email' => $data['email'],
        'password' => Hash::make($data['password']),
    ]);
    Frw_Utenti::create([
        'username' => $data['username'],
        'nome' => $data['name'],
        'cognome' => $data['name'],
        'password' => Hash::make($data['password']),
    ]);

    return $user;
    }

**Comment by trengram** [comment]
It is okay to have the users in AdAdmin tables. If there is a way to integrate with external login, the better. This will expand how many of us can integrate AdAdmin in our websites.

**Comment by giuliopons** [comment]
I've modified some files in AdAdmin installation to handle an external login process. It would be great if we can take a test in your installation. Can you please send me privately at giulio@barattalo.it an ftp access to your system? We can continue on email.

**Comment by giuliopons** [comment]
More info and detailed instructions here: #49
